interface/billing/ub04_dispose.php

```diff
@@ -0,0 +1,433 @@
+        $action = $_REQUEST['action'];
+        $ub04id = json_decode($ub04id, true);
+        saveTemplate($encounter, $pid, $ub04id, $action);
+        exit();
+    } elseif ($dispose == "payer_save") {
+        $ub04id = isset($_POST['ub04id']) ? $_POST['ub04id'] : $_GET['ub04id'];
+        $payerid = isset($_POST['payerid']) ? $_POST['payerid'] : $_GET['payerid'];
+        savePayerTemplate($payerid, $ub04id);
+        exit("done");
+    } elseif ($dispose == "batch_save") {
+        $pid = isset($_POST['pid']) ? $_POST['pid'] : $_GET['pid'];
+        $encounter = isset($_POST['encounter']) ? $_POST['encounter'] : $_GET['encounter'];
+        $ub04id = isset($_POST['ub04id']) ? $_POST['ub04id'] : $_GET['ub04id'];
+        saveTemplate($encounter, $pid, $ub04id, $dispose);
+        exit("done");
+    } elseif ($dispose == "reset_claim") {
+        $pid = isset($_POST['pid']) ? $_POST['pid'] : $_GET['pid'];
+        $encounter = isset($_POST['encounter']) ? $_POST['encounter'] : $_GET['encounter'];
+        // clear claim first otherwise get ub04 returns cuurent version.
+        //
+        $flg = exist_ub04_claim($pid, $encounter, true);
+        if ($flg === true) {
+            updateClaim(false, $pid, $encounter, - 1, - 1, - 1, - 1, '', 'ub04', - 1, 0, "");
+        }
+        $ub04id = get_ub04_array($pid, $encounter);
+        $ub04id = json_encode($ub04id);
+
+        echo $ub04id;
+        exit();
+    }
+    die(xl('Do not know what to do!'));
+}
+
+function get_payer_defaults($payerid)
+{
+    $p = sqlQuery("SELECT insurance_companies.* FROM insurance_companies WHERE insurance_companies.id = ?", array($payerid));
+    if ($p['claim_template']) {
+        return $p['claim_template'];
+    } else {
+        return json_encode(array());
+    }
+}
+
+function savePayerTemplate($payerid, $ubo4id)
+{
+    $ub04id = json_encode($ub04id);
+    sqlStatement("update insurance_companies set claim_template= ? where id = ?", array(
+        $ubo4id,
+        $payerid
+    ));
+}
+
+function saveTemplate($encounter, $pid, $ub04id, $action = 'form')
+{
+    if ($action != 'batch_save') {
+        $ub04id = json_encode($ub04id);
+        $isAuthorized = true;
+        ob_start();
+        include(dirname(__file__) . "/ub04_form.php");
+        $htmlin = ob_get_clean();
+        $isAuthorized = false;
+        ub04Dispose('download', $htmlin, "ub04_download.pdf", $action);
+    }
+    $flg = exist_ub04_claim($pid, $encounter, true);
+    if ($flg === true) {
+        updateClaim(false, $pid, $encounter, - 1, - 1, - 1, - 1, '', 'ub04', - 1, 0, $ub04id);
+    } else {
+        updateClaim(true, $pid, $encounter, - 1, - 1, 1, 1, '', 'ub04', - 1, 0, $ub04id);
+    }
+}
+
+function buildTemplate($pid = '', $encounter = '', $htmlin, $action = 'form', &$log)
+{
+    $ub04id = get_ub04_array($pid, $encounter, $log);
+
+    $ub04id = json_encode($ub04id);
+
+    $isAuthorized = true;
+    ob_start();
+    include(dirname(__file__) . "/ub04_form.php");
```

Comment:
Sometimes we don't want to stop a script on a dependency that will show itself in a more peaceful manner later. Still I changed.